site/gatsby-site/src/components/checklists/CheckListForm.js

```diff
@@ -0,0 +1,481 @@
+      <span {...{ className }}>
+        <Spinner />
+        <Trans>Saving...</Trans>
+      </span>
+    );
+  }
+  if (saveStatus == 'saved') {
+    return (
+      <span {...{ className }}>
+        <Trans>Saved</Trans>
+      </span>
+    );
+  }
+  if (saveStatus == 'error') {
+    return (
+      <span {...{ className }}>
+        <Spinner />
+        <Trans>Error saving...</Trans>
+      </span>
+    );
+  }
+}
+
+function Info({ children, className }) {
+  const [hide, setHide] = useState(false);
+
+  if (hide) return <></>;
+  return (
+    <div
+      className={`${className} bg-amber-50 text-amber-900 border border-amber-300 p-4 rounded shadow-md my-4`}
+    >
+      <button
+        className="border-0 bg-none float-right text-xl pl-4 mt-0 pb-2 -mt-2"
+        onClick={() => setHide(true)}
+      >
+        ×
+      </button>
+      <strong>INFO</strong>: {children}
+    </div>
+  );
+}
+
+// TODO: Group known and potential in GMF Taxonomy
+const searchRisks = async ({
+  values,
+  setFieldValue,
+  setRisksLoading,
+  setAllPrecedents,
+  setSaveStatus,
+}) => {
+  const queryTags = [...values['tags_goals'], ...values['tags_methods'], ...values['tags_other']];
+
+  if (queryTags.length == 0) return;
+
+  setRisksLoading(true);
+  const response = await fetch(
+    '/api/riskManagement/v1/risks?tags=' + encodeURIComponent(queryTags.join('___'))
+  );
+
+  const allPrecedents = [];
+
+  if (response.ok) {
+    const results = await response.json();
+
+    const risksToAdd = [];
+
+    for (let i = 0; i < results.length; i++) {
+      const result = results[i];
+
+      const newRisk = {
+        ...emptyRisk(),
+        title: abbreviatedTag(result.tag),
+        tags: [result.tag],
+        precedents: result.precedents,
+        description: result.description,
+        startClosed: true,
+      };
+
+      const notDuplicate = [...risksToAdd, ...values.risks].every(
+        (existingRisk) => !areDuplicates(existingRisk, newRisk)
+      );
+
+      if (notDuplicate) {
+        risksToAdd.push(newRisk);
+      }
+      for (const precedent of result.precedents) {
+        if (allPrecedents.every((p) => p.incident_id != precedent.incident_id)) {
+          allPrecedents.push(precedent);
+        }
+      }
+    }
+    setFieldValue('risks', values.risks.concat(risksToAdd));
+    setAllPrecedents(allPrecedents);
+
+    // Example result:
+    // [ { "tag": "GMF:Failure:Gaming Vulnerability",
+    //     "precedents": [
+    //       { "incident_id": 146,
+    //         "url": "https://incidentdatabase.ai/cite/146",
+    //         "title": "Research Prototype AI, Delphi, Reportedly Gave Racially Biased Answers on Ethics",
+    //         "description": "A publicly accessible research model[...]moral judgments.",
+    //         "tags": [ "GMF:Known AI Technology:Language Modeling", ],
+    //         "risk_tags": [ "GMF:Known AI Technical Failure:Distributional Bias", ]
+    //       },
+    //     ]
+    //   }
+    // ]
```

Comment:
Please remove the commented code or add a note explaining the reason to keep it.

Reply:
That wasn't commented code. It was an example of what the function might set `values.risks` to. I think that in general, examples are the easiest way to communicate what a piece of code does. However, I've moved it to the top of the function since that's probably where people are more likely to look.